packages/gestalt/src/Image.js

```diff
@@ -22,126 +23,107 @@ type Props = {|
   srcSet?: string,
 |};
 
-export default class Image extends PureComponent<Props> {
-  static propTypes = {
-    alt: PropTypes.string.isRequired,
-    children: PropTypes.node,
-    color: PropTypes.string,
-    fit: PropTypes.oneOf(['contain', 'cover', 'none']),
-    importance: PropTypes.oneOf(['high', 'low', 'auto']),
-    loading: PropTypes.oneOf(['eager', 'lazy', 'auto']),
-    naturalHeight: PropTypes.number.isRequired,
-    naturalWidth: PropTypes.number.isRequired,
-    onError: PropTypes.func,
-    onLoad: PropTypes.func,
-    sizes: PropTypes.string,
-    src: PropTypes.string.isRequired,
-    srcSet: PropTypes.string,
-  };
-
-  static defaultProps: {|
-    color: string,
-    fit?: 'contain' | 'cover' | 'none',
-    importance?: 'high' | 'low' | 'auto',
-    loading?: 'eager' | 'lazy' | 'auto',
-  |} = {
-    color: 'transparent',
-    fit: 'none',
-    importance: 'auto',
-    loading: 'auto',
-  };
-
-  componentDidMount() {
-    if (shouldScaleImage(this.props.fit)) {
-      this.loadImage();
+export default function Image({
+  alt,
+  children,
+  color = 'transparent',
+  fit = 'none',
+  importance = 'auto',
+  loading = 'auto',
+  naturalHeight,
+  naturalWidth,
+  onError,
+  onLoad,
+  sizes,
+  src,
+  srcSet,
+}: Props): Node {
+  const handleLoad: () => void = useCallback(() => {
+    if (onLoad) {
+      onLoad();
     }
-  }
+  }, [onLoad]);
 
-  componentDidUpdate(prevProps: Props) {
-    const { fit, src } = this.props;
-    if (shouldScaleImage(fit) && prevProps.src !== src) {
-      this.loadImage();
+  const handleError: () => void = useCallback(() => {
+    if (onError) {
+      onError();
     }
-  }
+  }, [onError]);
 
-  handleLoad: () => void = () => {
-    if (this.props.onLoad) {
-      this.props.onLoad();
-    }
-  };
-
-  handleError: () => void = () => {
-    if (this.props.onError) {
-      this.props.onError();
-    }
-  };
-
-  loadImage() {
+  const loadImage = useCallback(() => {
     if (typeof window !== 'undefined') {
       const image = new window.Image();
-      image.onload = this.handleLoad;
-      image.onerror = this.handleError;
-      image.src = this.props.src;
+      image.onload = handleLoad;
+      image.onerror = handleError;
+      image.src = src;
     }
-  }
+  }, [handleError, handleLoad, src]);
+
+  const previousSrc = usePrevious(src);
```

Comment:
useRef used to store info and replace componentDidUpdate

Reply:
That's correct - as explained in the React docs: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state